packages/syntax/src/renderingPlugin.tsx

```diff
@@ -132,61 +195,71 @@ export function treeToLines(
     }
   });
 
-  // Strip empty lines from the beginning of code blocks
-  while (lines[0].length === 0) {
-    lines.shift();
-  }
-
-  // Strip empty lines from the end of code blocks
-  while (lines[lines.length - 1].length === 0) {
-    lines.pop();
-  }
-
   return lines;
 }
 
-interface TableRowLineProps {
-  lineNumber: number;
-  line: Array<TreeItem>;
+interface TableContentProps {
+  lines: Array<Array<TreeItem>>;
 }
 
-function TableRowLine({ lineNumber, line }: TableRowLineProps) {
-  return (
-    <LineTableRow lineNumber={lineNumber}>
-      {line.map(processToken)}
-    </LineTableRow>
-  );
-}
+export function TableContent({ lines }: TableContentProps) {
+  const { highlightLines, showLineNumbers, darkMode } = useSyntaxContext();
+  const trimmedLines = lines;
 
-interface TableRowFragmentProps {
-  line: Array<TreeItem>;
-}
+  // Strip empty lines from the beginning of code blocks
+  while (trimmedLines[0].length === 0) {
+    lines.shift();
+  }
+
+  // Strip empty lines from the end of code blocks
+  while (trimmedLines[trimmedLines.length - 1].length === 0) {
+    trimmedLines.pop();
+  }
 
-function TableRowFragment({ line }: TableRowFragmentProps) {
   return (
     <>
-      <span>{line.map(processToken)}</span>
-
-      {/* We use a new line character here instead of <br /> so that text will not break outside of a <pre /> tag */}
-      {'\n'}
+      {trimmedLines.map((line, index) => {
+        const currentLineNumber = index + 1;
+        const highlightLine = highlightLines.includes(currentLineNumber);
+
+        let displayLineNumber;
+
+        if (showLineNumbers) {
+          displayLineNumber = currentLineNumber;
+        }
+
+        const processedLine = line.length ? (
+          line.map(processToken)
+        ) : (
+          // We create placeholder content when a line break appears to preserve the line break's height
+          // It needs to be inline-block for the table row to not collapse.
+          <div
+            className={css`
+              display: inline-block;
+            `}
+          />
+        );
+
+        return (
+          <LineTableRow
+            key={index}
```

Comment:
can we use `currentLineNumber` instead? no difference now, but would be more logical in a scenario where we could render a subset of all lines

Reply:
Given that `currentLineNumber` is just `index + 1`, it's going to suffer from the same issue as using index in this current implementation. I'm fine with the change though.